`initmake_asym` flags an error

exowanderer opened the pull request:
Hi Nate,

Your "make_asym.c" flags an error when I try to run "python setup.py install" or "./build.sh" (almost the same thing).

The error is:

```
make_asym.c:139:3: error: void function 'initmake_asym' should not return a value [-Wreturn-type]
return Py_InitModule("make_asym", make_asym_methods);
```

So I changed the function slightly to the following, and now it works without a hitch.

``` c
void
initmake_asym(void)
{
Py_InitModule("make_asym", make_asym_methods);
import_array();
}
```

I found this code snippet elsewhere that led me to the answer. It involved creating a macro for the condition "python3" vs "python2" situation. Therefore, the code that I am actually using looks like this:

``` c
#if PY_MAJOR_VERSION >= 3
#define NUMPY_IMPORT_ARRAY_RETURN_TYPE int
#else
#define NUMPY_IMPORT_ARRAY_RETURN_TYPE void

#endif

NUMPY_IMPORT_ARRAY_RETURN_TYPE
initmake_asym(void)
{
Py_InitModule("make_asym", make_asym_methods);
import_array();
}
```

exowanderer committed:
`initmake_asym` flags an error

Hi Nate,

Your "make_asym.c" flags an error when I try to run "python setup.py install" or "./build.sh" (almost the same thing).

The error is:
"""
make_asym.c:139:3: error: void function 'initmake_asym' should not return a value [-Wreturn-type]
return Py_InitModule("make_asym", make_asym_methods);
"""

So I changed the function slightly to the following, and now it works without a hitch.

void
initmake_asym(void)
{
Py_InitModule("make_asym", make_asym_methods);
import_array();
}

I found this code snippet elsewhere that led me to the answer. It involved creating a macro for the condition "python3" vs "python2" situation. Therefore, the code that I am actually using looks like this:

#if PY_MAJOR_VERSION >= 3
#define NUMPY_IMPORT_ARRAY_RETURN_TYPE int
#else
#define NUMPY_IMPORT_ARRAY_RETURN_TYPE void

#endif

NUMPY_IMPORT_ARRAY_RETURN_TYPE
initmake_asym(void)
{
Py_InitModule("make_asym", make_asym_methods);
import_array();
}

natelust commented on make_asym.c:
```diff
@@ -126,14 +126,19 @@ PyInit_make_asym(void)
+Py_InitModule("make_asym", make_asym_methods);
+import_array();
 }
-#endif
```
Removing this line causes a failure to compile. The above #if must be concluded at some point. If it is not desirable to have it here, it should be placed somewhere else.

exowanderer replied:
I thought that this was covered by adding an `#endif` exactly below the #define statement.  When I compile this code, I do not receive an error at this line. 

I will now add the `#endif` back into the code segment, but if it results in future errors, we know where to look.

exowanderer committed:
Included a final `#endif` below the code segment

Added `#define` lines to control for Python3 vs Python2 specific syntax

I was having trouble compiling this routine using the standard approached "python setup.py install". When I googled the error in question, I received information that led me to know it was a python versioning issue.  Specifically, Python3 requires the init module to return an int, while Python2 requires the init module to return a void.  By adding the `#define` lines, we are able to control for this depending on the version itself.

- Resolves compile issues with Python2, but controlling the init function declaration return
- Adds 5 lines to the entire code
- Final `#endif` -- after the `initmake_asym` function -- were requested by package maintainer: natelust
